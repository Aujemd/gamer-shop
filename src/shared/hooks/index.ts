export { useStoredCart } from "./use-stored-cart";
