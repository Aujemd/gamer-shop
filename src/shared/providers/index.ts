export * from "./data-fetching-provider";
