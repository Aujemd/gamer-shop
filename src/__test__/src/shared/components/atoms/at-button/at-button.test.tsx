import { fireEvent, render, screen } from "@testing-library/react";

import { AtButton } from "@/shared";

describe("atButton Component", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<AtButton>Click me</AtButton>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("applies custom className", () => {
        render(<AtButton className="custom-class" data-testid="button-test-id">Test</AtButton>);
        expect(screen.getByTestId("button-test-id")).toHaveClass("custom-class");
    });

    it("passes additional props to the button", () => {
        render(<AtButton data-testid="my-btn">Test</AtButton>);
        expect(screen.getByTestId("my-btn")).toBeInTheDocument();
    });

    it("calls onClick when clicked", () => {
        const handleClick = jest.fn();
        render(<AtButton onClick={handleClick} data-testid="button-test-id">Click</AtButton>);
        fireEvent.click(screen.getByTestId("button-test-id"));
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
