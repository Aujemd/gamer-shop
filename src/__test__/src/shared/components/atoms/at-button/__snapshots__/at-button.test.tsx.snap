// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atButton Component should render correctly 1`] = `
<DocumentFragment>
  <button
    class="font-archivo font-bold text-base leading-4 tracking-sm text-center rounded-lg "
  >
    Click me
  </button>
</DocumentFragment>
`;
